packages/server/src/sdk/app/rows/search.ts

```diff
@@ -92,12 +96,15 @@ export async function search(
       // Enrich saved query with ephemeral query params.
       // We prevent searching on any fields that are saved as part of the query, as
       // that could let users find rows they should not be allowed to access.
-      let viewQuery = dataFilters.buildQueryLegacy(view.query) || {}
-      delete viewQuery?.onEmptyFilter
+      let query = await enrichSearchContext(view.query || {}, context)
+      query = dataFilters.buildQueryLegacy(query) || {}
+      query = checkFilters(table, query)
```

Comment:
This `checkFilters` was extracted from within `searchInputMapping` for the sole purpose of being used here to correct user IDs.